networkx/algorithms/link_prediction.py

```diff
@@ -592,16 +592,57 @@ def predict(u, v):
 @not_implemented_for("directed")
 @not_implemented_for("multigraph")
 def katz_index(G, ebunch=None, beta=.1):
-    def predict(u, v):
-        if not nx.has_path(G, u, v):
-            return 0
-        A = nx.adjacency_matrix(G)
-        AB = A.multiply(beta)
-        I = sparse.identity(AB.shape[0])
-        result = sparse.linalg.spsolve(I - AB, I) - I
-        return result[u, v]
+    r"""Compute the Katz index of all node pairs in ebunch.
 
-    return _apply_prediction(G, predict, ebunch)
+    Katz index of nodes `u` and `v` is defined as
+
+    .. math::
+
+        S^{Katz}_{u, v} \(I - \beta A\)^{-1} - I
+
+    where A is the adjacency matrix of G and I is the identity matrix.
+    $\beta$ Provides an exponential damping factor such that shorter paths are valued more.
+
+    Parameters
+    ----------
+    G : graph
+        A NetworkX undirected graph.
+
+    ebunch : iterable of node pairs, optional (default = None)
+        Katz Index will be computed for each pair of nodes
+        given in the iterable. The pairs must be given as 2-tuples
+        (u, v) where u and v are nodes in the graph. If ebunch is None
+        then all non-existent edges in the graph will be used.
+        Default value: None.
+
```

Comment:
Maybe add a tick for the `beta` parameter. You could insist on the comment made in the [paper ](https://arxiv.org/pdf/1010.0725v1.pdf) that _"β must be lower than the reciprocal of the largest eigenvalue of
matrix A to ensure the convergence of the equation"_